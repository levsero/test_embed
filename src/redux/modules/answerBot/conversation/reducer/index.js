import { combineReducers } from 'redux';

import lastScroll from './last-scroll';
import lastScreenClosed from './last-screen-closed';

export default combineReducers({
  lastScroll,
  lastScreenClosed
});
