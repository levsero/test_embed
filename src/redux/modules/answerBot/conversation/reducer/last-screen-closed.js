import { CONVERSATION_SCREEN_CLOSED } from '../action-types'

const initialState = 0

const lastScreenClosed = (state = initialState, action) => {
  switch (action.type) {
    case CONVERSATION_SCREEN_CLOSED:
      return action.payload
    default:
      return state
  }
}

export default lastScreenClosed
