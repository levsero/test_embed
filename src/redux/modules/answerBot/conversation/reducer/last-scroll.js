import { CONVERSATION_SCROLL_CHANGED } from '../action-types';

const initialState = 0;

const lastScroll = (state = initialState, action) => {
  switch (action.type) {
    case CONVERSATION_SCROLL_CHANGED:
      return action.payload;
    default:
      return state;
  }
};

export default lastScroll;
