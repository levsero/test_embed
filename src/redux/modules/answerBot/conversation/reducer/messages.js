import {
  QUESTION_SUBMITTED_PENDING,
  QUESTION_SUBMITTED_FULFILLED
} from '../action-types';

import {
  BOT_MESSAGE,
  BOT_CHANNEL_CHOICE,
  BOT_FEEDBACK,
  BOT_FEEDBACK_REQUESTED
} from '../../root/action-types';

const initialState = new Map();

let messageCount = 0;

const addMessage = (messages, message) => {
  const copy = new Map(messages);

  return copy.set(messageCount++, { ...message });
};

const messages = (state = initialState, action) => {
  const { type, payload } = action;

  switch (type) {
    case QUESTION_SUBMITTED_PENDING:
      return addMessage(state, { ...payload, isVisitor: true });
    case QUESTION_SUBMITTED_FULFILLED:
      const {
        message, // eslint-disable-line no-unused-vars
        ...rest
      } = payload;

      return addMessage(state, { ...rest, type: 'results' });
    case BOT_MESSAGE:
      return addMessage(state, payload);
    case BOT_CHANNEL_CHOICE:
      return addMessage(state, { ...payload, type: 'channelChoice' });
    case BOT_FEEDBACK:
      return addMessage(state, { ...payload, isVisitor: true, type: 'feedback' });
    case BOT_FEEDBACK_REQUESTED:
      return addMessage(state, { ...payload, type: 'feedbackRequested' });
    default:
      return state;
  }
};

export default messages;
