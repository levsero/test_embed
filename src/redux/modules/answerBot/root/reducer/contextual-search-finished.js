import { CONTEXTUAL_SEARCH_FINISHED } from '../action-types';

const initialState = false;

const contextualSearchFinished = (state = initialState, action) => {
  switch (action.type) {
    case CONTEXTUAL_SEARCH_FINISHED:
      return true;
    default:
      return state;
  }
};

export default contextualSearchFinished;
