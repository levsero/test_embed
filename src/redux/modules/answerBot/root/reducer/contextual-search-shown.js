import { BOT_CONTEXTUAL_SEARCH_RESULTS } from '../action-types';

const initialState = false;

const contextualSearchShown = (state = initialState, action) => {
  switch (action.type) {
    case BOT_CONTEXTUAL_SEARCH_RESULTS:
      return true;
    default:
      return state;
  }
};

export default contextualSearchShown;
