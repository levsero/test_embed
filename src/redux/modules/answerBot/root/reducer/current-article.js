import { ARTICLE_SHOWN } from '../action-types'

const initialState = null

const currentArticle = (state = initialState, action) => {
  switch (action.type) {
    case ARTICLE_SHOWN:
      return {
        articleID: action.payload.articleID,
        sessionID: action.payload.sessionID
      }
    default:
      return state
  }
}

export default currentArticle
