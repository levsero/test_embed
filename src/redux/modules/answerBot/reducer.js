import { combineReducers } from 'redux';

import messages from './conversation/reducer/messages';
import conversation from './conversation/reducer';
import sessions from './sessions/reducer/sessions';
import currentMessage from './root/reducer/current-message';
import currentSessionID from './root/reducer/current-session-id';
import currentArticle from './root/reducer/current-article';
import currentContextualArticle from './root/reducer/current-contextual-article';
import currentScreen from './root/reducer/current-screen';
import greeted from './root/reducer/greeted';
import contextualSearchFinished from './root/reducer/contextual-search-finished';
import initialFallbackSuggested from './root/reducer/initial-fallback-suggested';

export default combineReducers({
  messages,
  conversation,
  currentMessage,
  sessions,
  currentSessionID,
  currentArticle,
  currentScreen,
  greeted,
  initialFallbackSuggested,
  contextualSearchFinished,
  currentContextualArticle
});
