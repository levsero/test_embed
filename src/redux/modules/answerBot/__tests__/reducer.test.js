import reducer from 'src/redux/modules/answerBot/reducer'

describe('answerBot root reducer', () => {
  it('has the expected sub states', () => {
    const state = reducer({}, { type: '' })

    expect(state).toMatchSnapshot()
  })
})
