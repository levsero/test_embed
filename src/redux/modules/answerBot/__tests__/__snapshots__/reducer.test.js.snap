// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`answerBot root reducer has the expected sub states 1`] = `
Object {
  "contextualSearchFinished": false,
  "conversation": Object {
    "getInTouchVisible": false,
    "lastScreenClosed": 0,
    "lastScroll": 0,
  },
  "currentArticle": null,
  "currentContextualArticle": null,
  "currentMessage": "",
  "currentScreen": "conversation",
  "currentSessionID": null,
  "greeted": false,
  "initialFallbackSuggested": false,
  "messages": Map {},
  "questionValueChangedTime": null,
  "sessions": Map {},
}
`;
